crates/sui-graphql-rpc/src/extensions/timeout.rs

```diff
@@ -74,13 +89,18 @@ impl Extension for TimeoutExt {
                     %error_code,
                     %query
                 );
-                Response::from_errors(vec![ServerError::new(
+                let error_msg = if *self.is_mutation.lock().unwrap() {
```

Comment:
nit: read the contents of `is_mutation` once and then re-use it rather than repeatedly locking.